Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,1111 @@
+    fMiscHeight[3] = 0.13392857f * fWideScreenHeightScale;
+    fMiscHeight[4] = 0.6473214f * fWideScreenHeightScale;
+    fMiscHeight[5] = 0.22321428f * fWideScreenHeightScale;
+    fMiscHeight[6] = 0.24553572f * fWideScreenHeightScale;
+    fMiscHeight[7] = 0.24553572f * fWideScreenHeightScale;
+    fMiscHeight[8] = 0.22321428f * fWideScreenHeightScale;
+    fMiscHeight[9] = 0.77678573f * fWideScreenHeightScale;
+    fMiscHeight[10] = 0.75446427f * fWideScreenHeightScale;
+    fMiscHeight[11] = 0.75446427f * fWideScreenHeightScale;
+    fMiscHeight[12] = 0.77678573f * fWideScreenHeightScale;
+    fMiscHeight[13] = 0.037946429f * fWideScreenHeightScale;
+    fMiscHeight[14] = 0.002232143f * fWideScreenHeightScale;
+    fMiscHeight[15] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[16] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[17] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[18] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[19] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[20] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[21] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[22] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[23] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[24] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[25] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[26] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[27] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fMiscHeight[28] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+}
+
+void UpdateScriptFixes()
+{
+    float w;
+    if (*pfScreenAspectRatio < fDefaultWidth)
+        w = static_cast<float>(pRsGlobal->width);
+    else
+        w = pRsGlobal->width * fDefaultWidth / *pfScreenAspectRatio;
+
+    fDefaultCoords = 0.5f * (pRsGlobal->width - w);
+}
+
+void UpdateHUDFixes()
+{
+    fRadarWidth[0] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[1] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[2] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[3] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[4] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[5] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[6] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[7] = 0.0015625f * fWideScreenWidthScale * fRadarWidthScale;
+    fRadarWidth[8] = 0.0015625f * fWideScreenWidthScale * fHudWidthScale;
+    fRadarWidth[9] = 0.0015625f * fWideScreenWidthScale * fHudWidthScale;
+    fRadarWidth[10] = 0.0015625f * fWideScreenWidthScale * fHudWidthScale;
```

Comment:
You can declare these magic numbers (0.0015625f, 0.0015625f, 0.002232143f) etc. as a constant variable

Reply:
These magic numbers are also magic to me, I will try but there's a lot through the code

Reply:
Therefore, you do not need to document them. Just list the ones that are repeated many times as variables like "RadarWidth_Mult" etc

Reply:
those that repeated many times became two variables, WidthMult and HeightMult, the remaining still being a magic number on the code because they do not repeat more than twice